File: components/common/src/androidMain/kotlin/com/mirego/pilot/components/ui/type/PilotKeyboardType.kt
package com.mirego.pilot.components.ui.type

import androidx.compose.ui.text.input.KeyboardType
import com.mirego.pilot.components.InternalPilotComponentsApi
import com.mirego.pilot.components.type.PilotKeyboardType

@InternalPilotComponentsApi
public val PilotKeyboardType.composeValue: KeyboardType
    get() = when (this) {
        PilotKeyboardType.Default -> KeyboardType.Text
        PilotKeyboardType.Ascii -> KeyboardType.Ascii
        PilotKeyboardType.Number -> KeyboardType.Number
        PilotKeyboardType.Email -> KeyboardType.Email
        PilotKeyboardType.Password -> KeyboardType.Password
        PilotKeyboardType.NumberPassword -> KeyboardType.NumberPassword
        PilotKeyboardType.Phone -> KeyboardType.Phone
        PilotKeyboardType.URL -> KeyboardType.Uri
    }

File: components/ios/base/Types/PilotKeyboardType.swift
import Shared
import SwiftUI

extension PilotKeyboardType {
    public var uiKeyboardType: UIKeyboardType {
        switch self {
        case .default:
            return .default
        case .ascii:
            return .asciiCapable
        case .number:
            return .numberPad
        case .email:
            return .emailAddress
        case .password:
            return .default
        case .numberPassword:
            return .numberPad
        case .phone:
            return .phonePad
        case .url:
            return .URL
        }
    }
}
